### PR TITLE
Drag drop

### PR DESCRIPTION
**Pull request by sirherrbatka**
Ok, done. Ignore the tablemimedata, table and document changes as those changes does not effect logic.

### CODE REVIEW

File: apps/opencs/view/world/scriptedit.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+            void dragEnterEvent (QDragEnterEvent* event);
+
+            void dropEvent (QDropEvent* event);
+
+            void dragMoveEvent (QDragMoveEvent* event);
+    };
+}
+#endif // SCRIPTEDIT_H
+// kate: indent-mode cstyle; indent-width 4; replace-tabs on;
```
Comment (zinnschlag) [inline apps/opencs/view/world/scriptedit.hpp]: junk?

Comment (sirherrbatka) [inline apps/opencs/view/world/scriptedit.hpp]: Nope, this comment help kdevelop to handle formatting better.

**Comment by zinnschlag** [comment]
JournalInfo is definitely wrong. But the need Topic.

Also, are you going through the command system? Like with everything else we can't manipulate the model directly.

And you should check, if the ID contains odd stuff (like spaces or special characters or beginning with a number) and if that is the case you should put it into quotation marks.

File: apps/opencs/view/world/scriptedit.hpp (hunk 1 above)
Comment (zinnschlag) [inline apps/opencs/view/world/scriptedit.hpp]: Yes, but that is junk IMO. There could be easily several dozen different editors in use by various team members. If we add hints for every single one of them, we end up with a sprawling mess of stuff in each file, that is not code.

**Comment by sirherrbatka** [comment]
I will correct those tomorrow.

About commands I simply subclassed QTextEdit and I use Qt methods to insert 
text into the view. I do not use model at all. This seems to work just fine 
(including undo stack).

File: apps/opencs/view/world/scriptedit.hpp (hunk 1 above)
Comment (sirherrbatka) [inline apps/opencs/view/world/scriptedit.hpp]: ok, fine.

**Comment by zinnschlag** [comment]
Okay. I don't remember exactly how QTextEdit works, but if you get the commands on the undo stack then everything should be okay.

**Comment by sirherrbatka** [comment]
Ok, At this point this works. I put everything in the quote for the sake of simplicity, but If this is needed I can create some regexp rules for that.

**Comment by zinnschlag** [comment]
It would be nice to get rid of the quotation marks when they are not needed. Since we are not going into the RC phase yet (damn bug), you might as well work on that.

**Comment by sirherrbatka** [comment]
To be honest, I don't know when they have to be included but operating on my 
basic assumption I created extreamly simple regular expression to deal with 
this. It is 100% safe minimal whitelist. To work further on it:

Does id has to be in qoute if it starts with _ or other char? I certainly know 
that numbers are not allowed.
We don't need to quote ids like foo_bar?
What about other chars? Like: $%^&*?!-=+

Sorry, no MW scripting expirence here at all.

**Comment by zinnschlag** [comment]
Basically anything goes, if it is in quotation marks. If it is not in quotation marks the situation is a bit less clear, but if we are defensive about it, we can use this rule: May only include alphanumeric characters and underscores and may not start with a number.

**Comment by sirherrbatka** [comment]
Done.

**Pull request merged**